modular_citadel/code/modules/reagents/reagent container/hypospraymkii.dm

```diff
@@ -2,105 +2,105 @@
 #define HYPO_INJECT 1
 
 //A vial-loaded hypospray. Cartridge-based!
-/obj/item/reagent_containers/hypospray/mkii
+/obj/item/hypospray/mkii
 	name = "hypospray mk.II"
 	icon = 'modular_citadel/icons/obj/hypospraymkii.dmi'
 	icon_state = "hypo2"
-	var/list/allowed_containers = list(/obj/item/reagent_containers/glass/bottle/vial/small)
-	desc = "A new development from DeForest Medical, this new hypospray takes 30-unit vials as the drug supply for easy swapping."
-	volume = 0
-	amount_per_transfer_from_this = 5
-	possible_transfer_amounts = list(5,10,15)
+	desc = "A new development from DeForest Medical, this hypospray takes 30-unit vials as the drug supply for easy swapping."
+	w_class = WEIGHT_CLASS_TINY
+	var/list/allowed_containers = list(/obj/item/reagent_containers/glass/bottle/vial/tiny, /obj/item/reagent_containers/glass/bottle/vial/small)
 	var/mode = HYPO_INJECT
 	var/obj/item/reagent_containers/glass/bottle/vial/vial
-	var/loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/small
+	var/start_vial = /obj/item/reagent_containers/glass/bottle/vial/small
 	var/spawnwithvial = TRUE
-	var/start_vial = null
+	var/inject_wait = 50
```

Comment:
should put these into defines instead of magic numbers.
#define HYPO_WAIT_STANDARD = 50

this is a 5 second delay